File: 987-vertical-order-traversal-of-a-binary-tree/987-vertical-order-traversal-of-a-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> verticalTraversal(TreeNode* root) {
        
        if(root == NULL)
            return {};
        
        // int a = 0, mini = 1e9, maxi = -1e9;
        // dfs(root, a, mini, maxi);
        // int size = maxi - mini + 1;
        // mini = -mini;
        vector<vector<int>> ans;
        map<int,vector<int>> m;
        queue<pair<TreeNode*,int>> q;
        q.push(make_pair(root,0));
        while(!q.empty()) {
            int currlen = q.size();
            map<int,vector<int>> temp;
            while(currlen > 0) {
                TreeNode* curr = q.front().first;
                if(curr -> left != NULL) {
                    q.push(make_pair(curr -> left, q.front().second - 1));
                }
                if(curr -> right != NULL) {
                    q.push(make_pair(curr -> right, q.front().second + 1));
                }
                temp[q.front().second].push_back(curr -> val);
                q.pop();
                currlen--;
            }
            for(auto x : temp) {
                vector<int> bruh = x.second;
                sort(bruh.begin(),bruh.end());
                for(auto y : bruh) {
                    m[x.first].push_back(y);
                }
            }
        }
        for(auto x : m) {
            ans.push_back(x.second);
        }
        
        return ans;
        
    }
};

File: 987-vertical-order-traversal-of-a-binary-tree/987-vertical-order-traversal-of-a-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def verticalTraversal(self, root: Optional[TreeNode]) -> List[List[int]]:
        
        
        d = {}
        
        def dfs(root,row,col):
            if not root:
                return
            if col in d:
                if row in d[col]:
                    d[col][row].append(root.val)
                else:
                    d[col][row] = [root.val]
            else:
                d[col] = {}
                d[col][row] = [root.val]
                
            dfs(root.left,row+1,col-1)
            dfs(root.right,row+1,col+1)
            
        dfs(root,0,0)
        
        keys = list(d.keys())
        keys.sort()
        ans = []
        for key in keys:
            tk = list(d[key].keys())
            tk.sort()
            val = []
            for k in tk:
                val.extend(sorted(d[key][k]))
            ans.append(val)
            
        return ans